### ISSUE TITLE
Support fix IPs

### ISSUE THREAD

**Issue by zs-zs** [getcarina/feedback#58]
Is there any plan to support fix IPs?

So if you have to rebuild your cluster, as in doing a `carina rebuild`, you always keep the same IP addresses for your nodes.

**Comment by everett-toews** [comment]
Have you ever had a `carina rebuild` assign you new IPs? It shouldn't be doing that, see [What does the cluster rebuild action do?](https://getcarina.com/docs/reference/faq/#what-does-the-cluster-rebuild-action-do)

**Comment by zs-zs** [comment]
Maybe it was a misunderstanding? :) See [the forum](https://community.getcarina.com/t/what-if-i-need-a-fix-ip/150).

**Comment by everett-toews** [comment]
From [this comment](https://community.getcarina.com/t/what-if-i-need-a-fix-ip/150), I mentioned

>If you have to rebuild your cluster, as in doing a `carina rebuild`, you always keep the same IP addresses for your nodes.

I thought perhaps you had another need for fixed IPs.

**Comment by zs-zs** [comment]
> I thought perhaps you had another need for fixed IPs.

No, for me any IP is OK if it doesn't change over time (so I can safely use it for a domain).
Sorry, I misunderstood your comment, I thought that you are saying that a cluster rebuild *can* potentially change the IP of my nodes. Should we close this?

**Comment by everett-toews** [comment]
One issue you'll eventually run into is that if you want to migrate to a new cluster because you want to work with a new version of Docker, you'll wind up getting a new IP address on that new cluster. Here's [an example](https://community.getcarina.com/t/upgrading-docker-version-on-cluster/85/4) of someone who is running into this issue now.

What do you think?

**Comment by zs-zs** [comment]
Oh, yes. :( I saw that issue. It's true... Well, a fixed IP could be a workaround for this. Or...a cluster rebuild could upgrade to a newer version if it's specified?

**Comment by everett-toews** [comment]
Rebuilds don't upgrade you. You'd have to migrate to a new cluster to get a newer version.

**Comment by everett-toews** [comment]
We have an issue to write a [Cluster upgrade tutorial](https://github.com/getcarina/getcarina.com/issues/797).

**Comment by zs-zs** [comment]
Yes it's currently not, but maybe in the future? Or would it be too difficult to make it work?
I think this would be the most comfortable from a user point of view. A rebuild option to upgrade to a specific version / or to the latest.

**Comment by thomasem** [comment]
We've toyed many times with the idea of allowing a way to upgrade a running cluster; it's more complicated than it seems and counter to the overall goal in a lot of ways - the main concern is how it may affect the stability of a user's application.

We try to be hands-off after we deploy the cluster. The idea of upgrading a running cluster is counter to our trying to treat application deployments more like cattle as well. No matter how good the infrastructure may be, eventually, the application will have to be moved to a new cluster - better to assume that as a possibility now and automate accordingly.

The most appropriate solution to this sort of problem, from what I can tell, is a fixed/floating IP, which we are currently looking at how we can support in the future.

**Comment by zs-zs** [comment]
> from what I can tell, is a fixed/floating IP, which we are currently looking at how we can support in the future

Okay, I'm convinced. :) If it's easier & safer, then for me it's also good!

**Comment by andrefernandes** [comment]
Fixed IPs could be very useful for corporate customers who are (desperately) trying to move stuff into Carina. The easiest way to deal with corporate IT madness is trying to negotiate a few outbound IP-based "holes" on firewall and (may they burn in Hell) HTTPS proxies.

Not having to deal with a VM makes Carina so sweet and kinda moves out of the way 80% of those imaginary fears from 90's corporate IT.
